spinn_front_end_common/utility_models/data_speed_up_packet_gatherer_machine_vertex.py

```diff
@@ -302,81 +436,571 @@ def get_local_provenance_data(self):
+        lead_monitor.set_reinjection_packets(
+            placements, extra_monitor_cores, transceiver,
             point_to_point=False, multicast=False, nearest_neighbour=False,
             fixed_route=False)
 
         # Clear any outstanding packets from reinjection
-        extra_monitor_cores_for_router_timeout[0].clear_reinjection_queue(
-            transceiver, placements, extra_monitor_cores_for_router_timeout)
-
-        # set time out
-        extra_monitor_cores_for_router_timeout[0].set_router_time_outs(
-            15, 15, transceiver, placements,
-            extra_monitor_cores_for_router_timeout)
-        extra_monitor_cores_for_router_timeout[0].\
-            set_reinjection_router_emergency_timeout(
-                1, 1, transceiver, placements,
-                extra_monitor_cores_for_router_timeout)
-
-    def unset_cores_for_data_extraction(
-            self, transceiver, extra_monitor_cores_for_router_timeout,
-            placements):
+        lead_monitor.clear_reinjection_queue(
+            transceiver, placements, extra_monitor_cores)
+
+        # set time outs
+        lead_monitor.set_router_emergency_timeout(
+            1, 1, transceiver, placements, extra_monitor_cores)
+        lead_monitor.set_router_time_outs(
+            15, 15, transceiver, placements, extra_monitor_cores)
+
+    @staticmethod
+    def load_application_routing_tables(
+            transceiver, extra_monitor_cores, placements):
+        """ Set all chips to have application table loaded in the router
+
+        :param transceiver: the SpiNNMan instance
+        :param extra_monitor_cores: the extra monitor cores to set
+        :param placements: placements object
+        :rtype: None
+        """
+        extra_monitor_cores[0].load_application_mc_routes(
+            placements, extra_monitor_cores, transceiver)
+
+    @staticmethod
+    def load_system_routing_tables(
+            transceiver, extra_monitor_cores, placements):
+        """ Set all chips to have the system table loaded in the router
+
+        :param transceiver: the SpiNNMan instance
+        :param extra_monitor_cores: the extra monitor cores to set
+        :param placements: placements object
+        :rtype: None
+        """
+        extra_monitor_cores[0].load_system_mc_routes(
+            placements, extra_monitor_cores, transceiver)
+
+    def unset_cores_for_data_streaming(
+            self, transceiver, extra_monitor_cores, placements):
+        """ Helper method for setting the router timeouts to a state usable\
+            for data streaming
+
+        :param transceiver: the SpiNNMan instance
+        :param extra_monitor_cores: the extra monitor cores to set
+        :param placements: placements object
+        :rtype: None
+        """
+        lead_monitor = extra_monitor_cores[0]
+        # Set the routers to temporary values
+        lead_monitor.set_router_time_outs(
+            15, 4, transceiver, placements, extra_monitor_cores)
```

Comment:
This assumes that 15, 4 *was* the value that the router time outs was set to.  I think the code used to get the status from at least one core and then use that; given that the timeout is not explicitly set anywhere in the Python code, this could lead to some confusion if we decide to change this in the future in the C code...

Reply:
I don't know why @alan-stokes put that in; the Java code doesn't bother and doesn't seem to have a problem. (The real router timeouts are set just a little bit further down.)